spec/factories/journals.rb

```diff
@@ -3,5 +3,47 @@
     is_successful true
     created_by 1
     journal_date { Time.zone.now }
+
+    trait :with_completed_order do
+      facility
+
+      transient do
+        quantities [1]
+
+        facility_account do
+          facility
+            .facility_accounts
+            .create(FactoryGirl.attributes_for(:facility_account))
+        end
+
+        order { FactoryGirl.create(:purchased_order, product: product) }
+
+        product do
```

Comment:
Does `association :product, factory: :setup_item, facility: facility, facility_account: facility_account` work?

Similar for the `order` association. Seems like it might be a little cleaner.